instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py

```diff
@@ -110,6 +111,19 @@ def _patched_api_call(self, original_func, instance, args, kwargs):
         error = None
         result = None
 
+        # inject trace context into payload headers for lambda Invoke
+        if (
+            service_name == "lambda"
+            and operation_name == "Invoke"
+            and "Payload" in api_params
+        ):
```

Comment:
*If* there is a way, I think we should be patching the lamba invoke function or another place before this func directly to inject context. deserializing and then re-serializing JSON doesn't feel right.

Reply:
I don't think it's possible - according to boto doc,`Payload (bytes or seekable file-like object) -- The JSON that you want to provide to your Lambda function as input.` So either way we aren't going to get a dict, even at the entry point.
We could avoid (de)serializing JSON by using pure string manipulation (find headers by string, deserialize, modifu, serialize again, inject). This however is more tricky and error-prone (but possibly much faster). @owais I defer the decision to you, I'm too newbie here :)

Reply:
I was thinking about raw string manipulation as well. In simple cases we probably only need to strip last character '}', concatinate our string and add back the last char but yea it can be error prone. 

I think parsing and dumping again is fine if that's the only way. 

BTW what if we inject context into HTTP request headers instead of the payload. Will lamba be able to read them?

Reply:
As far as I can tell (again, no expert here!) in order to get HTTP header values as part of the event in lambda handler, each customer would need to configure a mapping template for their lambda in API gateway (header val -> payload val). So I think that this would be too much of an overhead. Will go with raw string and see how it goes :)

Reply:
Makes sense.

Reply:
@owais pushed raw string version. In order to handle corner cases (existing headers prop in payload dict) I had to add quite a few lines. Looks solid but it's definitely more complex than straight json (de)serialization. Please have a look :)

Reply:
Oh you went all the way with raw string manipulation :) I can't say I really understand it at a quick glance haha :)
I'm not entirely sure if we should use this or the json module. Let's see what others have to say about it.

Curious if you benchmarked it and how faster it was compared to using json module?

Reply:
Frankly speaking I have not. I'll run some microbenchmarks with timeit to compare both implementations.

Reply:
@owais added some perf microbenchmarking results :)

Reply:
Switched back to JSON, as discussed elsewhere.